File: DSTU/hw2/app/src/main/java/ru/averveyko/hw2/SecondActivity.java
package ru.averveyko.hw2;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        String txt = getIntent().getStringExtra("txt");

        TextView txtMsg = findViewById(R.id.txt_msg);
        txtMsg.setText(txt);

        Button butShare = findViewById(R.id.but_share);
        butShare.setOnClickListener(v -> {
            Intent sendIntent = new Intent();
            sendIntent.setAction(Intent.ACTION_SEND);
            sendIntent.putExtra(Intent.EXTRA_TEXT, txt);
            sendIntent.setType("text/plain");

            Intent shareIntent = Intent.createChooser(sendIntent, null);
            startActivity(shareIntent);
        });
    }
}

File: AppOne/app/src/main/java/ru/averveyko/appone/SecondActivity.kt
package ru.averveyko.appone

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class SecondActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.second_activity)

        val button = findViewById<Button>(R.id.act2_button)
        val edit = findViewById<EditText>(R.id.act2_edit)

        button.setOnClickListener {
            val newText = edit.text.toString()
            val intent = Intent()
            intent.putExtra("txt", newText)
            setResult(0, intent)
            finish()
        }

        val stringExtra = intent.getStringExtra("txt")
        edit.setText(stringExtra)
    }
}